resources/js/shared/components/cities/CitiesTable.tsx

```diff
@@ -0,0 +1,226 @@
+import { useEffect, useState } from "react";
+
+import { deleteCity, getCities } from "~/api/cities";
+import { CreateCityModal } from "~/modals/City/CreateCityModal";
+import { EditCityModal } from "~/modals/City/EditCityModal";
+import { Button } from "~/ui";
+
+interface Pagination {
+  count: number;
+  total: number;
+  perPage: number;
+  currentPage: number;
+  totalPages: number;
+  links: {
+    next: string;
+    previous: string;
+  };
+}
+interface Filters {
+  [key: string]: string;
+}
+
+export const CitiesTable = () => {
+  const [loading, setLoading] = useState<boolean>(true);
```

Comment:
```suggestion
  const [loading, setLoading] = useState(true);
```

There is no need to specify the type here as typescript infers it's a boolean